sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,379 @@
+# TODO: Once we no longer support Python 2, we can eliminate this and get the
+# name using the 'name' field (e.g., `signal.SIGINT.name`) starting with Python 3.5
+SIGNALS_TO_NAMES_DICT = dict((getattr(signal, n), n)
+                             for n in dir(signal) if n.startswith('SIG') and '_' not in n)
+
+SYSLOG_IDENTIFIER = "stormond"
+
+STORAGE_DEVICE_TABLE = "STORAGE_INFO"
+FSSTATS_SYNC_TIME_KEY = "FSSTATS_SYNC"
+
+FSIO_RW_JSON_FILE = "/usr/share/stormond/fsio-rw-stats.json"
+
+STORMOND_MAIN_THREAD_SLEEP_SECS = 3600 #one hour
+STORMOND_SYNC_TO_DISK_SECS = 86400 #one day
+
+STORAGEUTIL_LOAD_ERROR = 127
+
+log = logger.Logger(SYSLOG_IDENTIFIER)
+
+exit_code = 0
+
+#
+# Daemon =======================================================================
+#
+
+
+class DaemonStorage(daemon_base.DaemonBase):
+
+    def __init__(self, log_identifier):
+        super(DaemonStorage, self).__init__(log_identifier)
+
+        self.timeout = STORMOND_MAIN_THREAD_SLEEP_SECS
+        self.fsstats_sync_interval = STORMOND_SYNC_TO_DISK_SECS
+        self.stop_event = threading.Event()
+        self.state_db = None
+        self.config_db = None
+        self.device_table = None
+        self.storage = StorageDevices(log_identifier)
+
+        # These booleans are for FSIO RW information reconciliation
+        self.fsio_json_file_loaded = False
+        self.use_fsio_json_baseline = False
+        self.statedb_storage_info_loaded = False
+        self.use_statedb_baseline = False
+
+        # These dicts are to load info from disk/database into memory, respectively
+        self.fsio_rw_json = dict((disk, {}) for disk in self.storage.devices)
+        self.fsio_rw_statedb = dict((disk, {}) for disk in self.storage.devices)
+
+        # This time is set at init and then subsequently after each FSIO JSON file sync
+        self.fsio_sync_time = time.time()
+
+        # these are the config timeout values we get from the config_db:
+        self.config_fields = ["daemon_polling_interval" , "fsstats_sync_interval"]
+        
+        # These are the various static and dynamic fields that are posted to state_db
+        self.static_fields = ["device_model", "serial"]
+        self.dynamic_fields = ["firmware", \
+                               "health", \
+                               "temperature", \
+                               "latest_fsio_reads", \
+                               "latest_fsio_writes", \
+                               "total_fsio_reads", \
+                               "total_fsio_writes", \
+                               "disk_io_reads", \
+                               "disk_io_writes", \
+                               "reserved_blocks"]
+
+        # These are the fields that we are interested in saving to disk to protect against
+        # reboots or crashes
+        self.statedb_json_sync_fields = self.dynamic_fields[3:7]
+
+        # Connect to CONFIG_DB and get polling and sync intervals
+        self._get_configdb_intervals()
```

Comment:
That is correct. User would need to set the parameters again and then restart the daemon in its current design iteration.

Reply:
Modified the logic so that _get_configdb_intervals() is called after each timeout period. This ensures that these intervals can be changed dynamically.